File: pinyougou_shop_web/src/main/java/com/pinyougou/shop/controller/LoginController.java
package com.pinyougou.shop.controller;

import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/login")
public class LoginController {

    //基于安全框架获取商家用户名
    @RequestMapping("/getName")
    public Map<String,String> getName(){
        String loginName = SecurityContextHolder.getContext().getAuthentication().getName();
        Map<String, String> map = new HashMap<>();
        map.put("loginName",loginName);
        return map;
    }
}

File: pinyougou_shop_web/src/main/webapp/js/controller/loginController.js
app.controller("loginController",function ($scope, $controller, loginService) {

    //继承父类控制器
    $controller("baseController",{$scope:$scope});

    //基于安全框架获取商家用户名
    $scope.getName = function () {
        loginService.getName().success(function (response) {
            $scope.loginName = response.loginName;
        })
    }
});